src/modules/item/sagas.spec.ts

```diff
@@ -617,3 +622,85 @@ describe('when publishing third party items', () => {
     })
   })
 })
+
+describe('when handling the downloadItemRequest action', () => {
+  let itemsById: Record<string, Item>
+
+  beforeEach(() => {
+    itemsById = {
+      male: { id: 'male', name: 'male', contents: { 'male/model.glb': 'Qmhash' } as Record<string, string> } as Item,
+      maleAndFemale: {
+        id: 'maleAndFemale',
+        name: 'male and female',
+        contents: { 'male/model_male.glb': 'QmhashMale', 'female/model_female.glb': 'QmhashFemale' } as Record<string, string>
+      } as Item,
+      both: {
+        id: 'both',
+        name: 'both',
+        contents: { 'male/model.glb': 'Qmhash', 'female/model.glb': 'Qmhash' } as Record<string, string>
+      } as Item
+    }
+  })
+
+  describe('when id is not found', () => {
+    const itemId = 'invalid'
+    it('should throw an error with a message that says the item was not found', () => {
+      return expectSaga(itemSaga, builderAPI)
+        .provide([[select(getItemsById), itemsById]])
+        .put(downloadItemFailure(itemId, `Item not found for itemId="invalid"`))
+        .dispatch(downloadItemRequest(itemId))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe('when an item has only one "male" representation', () => {
+    it('should download a zip file with all the contents under a /male directory', () => {
+      const itemId = 'male'
+      const files = { 'male/model.glb': 'Qmhash' }
+      return expectSaga(itemSaga, builderAPI)
+        .provide([
+          [select(getItemsById), itemsById],
+          [call([builderAPI, 'fetchContent'], 'Qmhash'), new Blob()],
+          [matchers.call.fn(buildZipFiles), files],
+          [call(downloadZip, 'male', files), undefined]
+        ])
+        .put(downloadItemSuccess(itemId))
+        .dispatch(downloadItemRequest(itemId))
+        .run({ silenceTimeout: true })
+    })
+  })
+
+  describe('when an item has two different representations for male and female', () => {
+    it('should download a zip file with both /male and /female directories', () => {
+      const itemId = 'maleAndFemale'
+      const files = { 'male/model_male.glb': 'QmhashMale', 'female/model_female.glb': 'QmhashFemale' }
+      return expectSaga(itemSaga, builderAPI)
+        .provide([
+          [select(getItemsById), itemsById],
+          [call([builderAPI, 'fetchContent'], 'Qmhash'), new Blob()],
+          [matchers.call.fn(buildZipFiles), files],
```

Comment:
It would have been great to have this matcher to do a deep equal comparison. We'll do it next time.